app/src/main/java/com/example/playlistmaker/SearchActivity.kt

```diff
@@ -52,58 +85,88 @@ class SearchActivity : AppCompatActivity() {
         }
         inputEditText.addTextChangedListener(simpleTextWatcher)
 
+
         // Обработка нажатия на поле ввода для отображения клавиатуры
         inputEditText.setOnClickListener {
-                inputEditText.text.clear()
-                showSoftKeyboard()
+            inputEditText.text.clear()
+            showSoftKeyboard()
             // Показ клавиатуры
+        }
+
+        udpateButton.setOnClickListener {
+            if (lastSearchQuery != null) {
+                performSearch(lastSearchQuery!!)
```

Comment:
Чтобы тут не использовать оператор `!!` нужно вынести отдельную переменную:
```suggestion
            val query = lastSearchQuery
            if (query  != null) {
                performSearch(query)
```

Reply:
Принял, спасибо